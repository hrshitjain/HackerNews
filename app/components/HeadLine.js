import { Text, TouchableHighlight, View } from "react-native"
import { HeadlinesStyle } from "../css/HeadLinesStyle";
import React, { memo } from 'react';

const HeadLine = memo(({title, onPress}) => {
    return(
    <TouchableHighlight onPress={onPress}>
        <View style={HeadlinesStyle.item}>
            <Text style={HeadlinesStyle.title}>{title}</Text>
        </View>
    </TouchableHighlight>);});

export default HeadLine;