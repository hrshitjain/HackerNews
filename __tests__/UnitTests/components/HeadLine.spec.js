import React from 'react';
import renderer from 'react-test-renderer';
import HeadLine from '../../../app/components/HeadLine';

describe('HeadLine component render', () =>{
    const component = renderer.create(<HeadLine/>).toJSON();

    it('snapshot test', () =>{
        expect(component).toMatchSnapshot();
    });
    
});