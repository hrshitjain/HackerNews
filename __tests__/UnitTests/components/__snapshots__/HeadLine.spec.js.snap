// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadLine component render snapshot test 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "marginHorizontal": 16,
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;
